packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1849,12 +1854,8 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
       ? null
       : _textPainter.getPositionForOffset(globalToLocal(to - _paintOffset));
 
-    int baseOffset = fromPosition.offset;
-    int extentOffset = fromPosition.offset;
-    if (toPosition != null) {
-      baseOffset = math.min(fromPosition.offset, toPosition.offset);
-      extentOffset = math.max(fromPosition.offset, toPosition.offset);
-    }
+    final int baseOffset = fromPosition.offset;
```

Comment:
This seems to revert the changes made in https://github.com/flutter/flutter/pull/29395. According to @mdebbar 

> The problem is if you drag from left to right, the text will be selected as expected. But if you drag from right to left, then baseOffset will be greater than extentOffset and it'll be collapsed down to a cursor. So dragging from right to left is broken, and this PR fixes it.

Not sure why the tests are passing. But it sounds like we're not handling baseOffset > extentOffset correctly?

Reply:
Ah thanks for tracking that PR down.  It seems like the bug that that fixed, where the cursor would collapsed when selecting right to left, no longer happens.  Everything still seems to be fine with this PR here by my testing.

I should point out that we're still reversing base and extent in the underlying TextRange.  That seems to have been around for a long time, though.

https://github.com/flutter/flutter/blob/72aa23eb935dc9d1d667163e7ffffcb1e7319de8/packages/flutter/lib/src/services/text_editing.dart#L18-L26